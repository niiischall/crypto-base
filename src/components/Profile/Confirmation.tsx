import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { useDispatch } from 'react-redux';
import { Button, withStyles } from '@material-ui/core';

import { authLogout } from '../../store/actions/actionProfile';

export interface Props {
  switchPage: Function;
}

const useStyles = makeStyles((theme) => ({
  container: {
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  image: {
    width: 200,
    height: 200,
  },
  heading: {
    fontSize: '20px',
    color: '#669900',
  },
  statement: {
    fontSize: '16px',
  },
  buttonContainer: {
    width: '100%',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 15,
  },
}));

const StyledButton = withStyles((theme) => ({
  root: {
    height: 45,
    marginTop: 15,
    marginRight: 10,
    width: '40%',
    borderRadius: '6px',
    color: '#fff',
    fontSize: '16px',
    textTransform: 'capitalize',
    backgroundColor: '#000',
    boxShadow: '0 0 0 1px rgb(0 0 0 / 5%), 0 5px 15px rgb(0 0 0 / 15%)',
    '&:hover': {
      color: '#000',
    },
  },
  endIcon: {
    margin: 0,
  },
}))((props: any) => <Button disableRipple {...props} />);

export const Confirmation: React.FC<Props> = (props) => {
  const classes = useStyles();
  const dispatch = useDispatch();

  const handleLogout = () => {
    dispatch(authLogout());
  };

  const handleNavigation = () => {
    props.switchPage('home');
  };

  return (
    <div className={classes.container}>
      <img
        className={classes.image}
        src="/img/high-five.svg "
        alt="high-five"
      />
      <h3 className={classes.heading}>You're in!</h3>
      <p className={classes.statement}>We're so glad you're here.</p>
      <div className={classes.buttonContainer}>
        <StyledButton onClick={handleNavigation}>Home</StyledButton>
        <StyledButton onClick={handleLogout}>Logout</StyledButton>
      </div>
    </div>
  );
};

export default Confirmation;
