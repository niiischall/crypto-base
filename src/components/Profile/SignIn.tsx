import React from 'react';
import { makeStyles } from '@material-ui/core/styles';

export interface Props {
  toggleForm: Function;
}

const useStyles = makeStyles((theme) => ({
  signInContainer: {
    height: '100%',
    flex: 1,
    padding: 30,
    backgroundColor: '#fff',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'flex-start',
    alignItems: 'center',
  },
  header: {
    width: '100%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
  heading: {
    fontSize: 22,
    lineHeight: '32px',
    fontWeight: 700,
    letterSpacing: '1px',
    marginRight: 'auto',
  },
  headerButton: {
    color: '#4A6EE0',
    fontSize: 14,
    lineHeight: '22px',
    backgroundColor: '#fff',
    border: 'none',
    textDecoration: 'underline',
    cursor: 'pointer',
  },
}));

export const SignIn: React.FC<Props> = ({ toggleForm }) => {
  const classes = useStyles();

  return (
    <div className={classes.signInContainer}>
      <div className={classes.header}>
        <span className={classes.heading}>Sign in</span>
        <button className={classes.headerButton} onClick={() => toggleForm()}>
          I don't have an account
        </button>
      </div>
    </div>
  );
};

export default SignIn;
