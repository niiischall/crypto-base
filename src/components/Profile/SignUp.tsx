import React from 'react';
import { makeStyles } from '@material-ui/core/styles';

export interface Props {
  toggleForm: Function;
}

const useStyles = makeStyles((theme) => ({
  signUpContainer: {
    height: '100%',
    padding: 30,
    backgroundColor: '#fff',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'flex-start',
    alignItems: 'center',
  },
  header: {
    width: '100%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
  heading: {
    fontSize: 22,
    lineHeight: '32px',
    fontWeight: 700,
    letterSpacing: '1px',
    marginRight: 'auto',
  },
  headerButton: {
    color: '#4A6EE0',
    fontSize: 14,
    lineHeight: '22px',
    backgroundColor: '#fff',
    border: 'none',
    textDecoration: 'underline',
    cursor: 'pointer',
  },
}));

export const SignUp: React.FC<Props> = ({ toggleForm }) => {
  const classes = useStyles();

  return (
    <div className={classes.signUpContainer}>
      <div className={classes.header}>
        <span className={classes.heading}>Sign up</span>
        <button className={classes.headerButton} onClick={() => toggleForm()}>
          I don't have an account
        </button>
      </div>
    </div>
  );
};

export default SignUp;
