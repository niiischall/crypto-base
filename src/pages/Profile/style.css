.profile-container {
  background-color: salmon;
  flex: 1;
}