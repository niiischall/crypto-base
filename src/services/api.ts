const baseCoinAPI = 'https://pro-api.coinmarketcap.com/';
const baseAuthAPI = 'https://identitytoolkit.googleapis.com/';
const apiVersion = 'v1';
const authVersion = 'v1';
const apiKey = `&CMC_PRO_API_KEY=${process.env.REACT_APP_CMC_KEY}`;
const authKey = `?key=${process.env.REACT_APP_AUTH_KEY}`;

export const apiEndPoints = {
  totalCoins:
    '/cryptocurrency/map?sort=cmc_rank&limit=500&listing_status=active',
  popularCoins: '/cryptocurrency/listings/latest?limit=25',
  trendingCoins:
    '/cryptocurrency/listings/latest?sort=percent_change_24h&limit=25',
  coinsInfo: '/cryptocurrency/info?slug=',
  coinsQuotes: '/cryptocurrency/quotes/latest?slug=',
  signUp: `/accounts:signUp`,
  signIn: `/accounts:signInWithPassword`,
};

export async function fetchApi(
  endPoint: string,
  payload?: {} | undefined,
  method: string = 'get',
  headers?: {},
) {
  let path = `${baseCoinAPI}${apiVersion}${endPoint}${apiKey}`;
  const headersObject: any = {
    ...headers,
  };
  let request = {
    body: JSON.stringify(payload),
    headers: headersObject,
    method: method.toLowerCase(),
  };

  try {
    return fetch(path, request);
  } catch (e: any) {
    const stringError = e && e.toString && e.toString();
    const type =
      stringError === 'TypeError: Network request failed'
        ? 'networkError'
        : 'unknown';
    const error = {
      text: stringError,
      type,
    };
    throw error;
  }
}

export async function authApi(
  endPoint: string,
  method: string,
  payload: {},
  headers?: {},
) {
  const path = `${baseAuthAPI}${authVersion}${endPoint}${authKey}`;
  const headersObject: any = {
    ...headers,
  };
  let request = {
    body: JSON.stringify(payload),
    headers: headersObject,
    method: method.toLowerCase(),
  };

  try {
    return fetch(path, request);
  } catch (e: any) {
    const stringError = e && e.toString && e.toString();
    const type =
      stringError === 'TypeError: Network request failed'
        ? 'networkError'
        : 'unknown';
    const error = {
      text: stringError,
      type,
    };
    throw error;
  }
}
