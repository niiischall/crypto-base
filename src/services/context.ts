import React from 'react';

//Page Values
export const pages = {
  home: 'home',
  search: 'search',
  profile: 'profile',
};

//Page context
const pageContext = React.createContext({
  page: '',
  switchPage: (event: any, value: string) => {},
});

//Page Provider
export const PageProvider = pageContext.Provider;

//Default pageContext Export
export default pageContext;
