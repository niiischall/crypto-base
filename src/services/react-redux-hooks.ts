import {
  useSelector as originalUseSelector,
  useDispatch as originalUseDispatch,
} from 'react-redux';

export const useSelector: any = (state: any) => originalUseSelector(state);
export const useDispatch: any = () => originalUseDispatch();
