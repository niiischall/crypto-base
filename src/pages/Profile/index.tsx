import React from 'react';
import './style.css';

export const Profile: React.FC<{}> = (props) => {
  return (
    <div className="profile-container">
      <span>&nbsp;</span>
    </div>
  );
};

export default Profile;
