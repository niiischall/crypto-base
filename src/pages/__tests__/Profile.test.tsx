import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import { store } from '../../mocks/mock';
import * as ReactReduxHooks from '../../services/react-redux-hooks';

import { Profile } from '../Profile';

const setup = () => {
  const component = mount(
    <Provider store={store}>
      <Profile />
    </Provider>,
  );
  return {
    component,
  };
};

describe('Profile Component', () => {
  let enzymeWrapper: any;
  const { component } = setup();

  beforeEach(() => {
    /* mocking useSelector on our mock store */
    jest
      .spyOn(ReactReduxHooks, 'useSelector')
      .mockImplementation((state) => store.getState());
    /* mocking useDispatch on our mock store  */
    jest
      .spyOn(ReactReduxHooks, 'useDispatch')
      .mockImplementation(() => store.dispatch);

    /* shallow rendering */
    enzymeWrapper = component;
  });

  test('should match snapshot', () => {
    expect(enzymeWrapper).toMatchSnapshot();
  });
});
