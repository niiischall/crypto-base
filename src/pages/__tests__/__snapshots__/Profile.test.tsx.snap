// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component should match snapshot 1`] = `ReactWrapper {}`;
